lib/spack/spack/build_environment.py

```diff
@@ -1218,8 +1262,8 @@ def _setup_pkg_and_run(
         elif context == "test":
             logfile = os.path.join(pkg.test_suite.stage, pkg.test_suite.test_log_name(pkg.spec))
 
-        error_msg = str(exc)
-        if isinstance(exc, (spack.multimethod.NoSuchMethodError, AttributeError)):
+        error_msg = str(e)
+        if isinstance(e, (spack.multimethod.NoSuchMethodError, AttributeError)):
```

Comment:
we might also want to delete this branch, the last few times i've hit it it was a false positive and only confusing

Reply:
can do in a separate PR